lts-test/src/main/java/com/google/cloud/tools/opensource/lts/GradleProjectModifier.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.lts;
+
+import com.google.cloud.tools.opensource.dependencies.Bom;
+import com.google.common.base.Charsets;
+import com.google.common.base.Verify;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Files;
+import com.google.common.io.MoreFiles;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/** Modifies build.gradle files to use the libraries in the BOM when running the unit tests. */
+class GradleProjectModifier implements BuildFileModifier {
+  @Override
+  public void modifyFiles(String name, Path projectDirectory, Bom bom)
+      throws TestSetupFailureException {
+    Iterable<Path> paths = MoreFiles.fileTraverser().breadthFirst(projectDirectory);
+    try {
+      for (Path path : paths) {
+        if (path.endsWith("build.gradle")) {
```

Comment:
This code is mixing OO and procedural approaches. It feels to me like there are really three different types of object here, one for each branch, that could be separate classes.

Reply:
Good point. Now BeamProjectModifier is an independent class.